postgresql_11: remove

Ma27 opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

This version will stop receiving fixes as of Nov 09 2023, so it needs to be removed before 23.11.

cc @NixOS/nixos-release-managers 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Ma27 committed:
postgresql_11: remove

As described in the release lifecycle docs from postgresql[1], v11 will
stop receiving fixes as of Nov 9 2023. This means it's EOL throughout
the entire lifetime of 23.11, so let's drop it now.

A lot of examples are also referencing postgresql_11. Where it's
sensible, use postgresql_15 as example now to avoid confusion.

This is also handy because the LLVM 16 fix for postgresql is not
available for postgresql 11 ;-)

[1] https://www.postgresql.org/support/versioning/

Ma27 committed:
glom: provide bin dir of postgresql

By default `pg_config --bindir` is executed during configure phase to
get the bin directory containing all the postgres executables used by
glom.

This seems wrong given that the same postgresql_15 package will be
executed on build and on the final host platform which is not
necessarily the same. To avoid that, specify the bin dir manually via
the corresponding configure option.

natsukium commented:
@ofborg eval

the-furry-hubofeverything reviewed (commented):
Should an entry be added in `pkgs/top-level/aliases.nix`?

@natsukium The eval is failing because https://github.com/NixOS/nixpkgs/pull/264290 isn't included, needs rebase

the-furry-hubofeverything commented:
What about postgresql_10?

Ma27 commented:
> Should an entry be added in pkgs/top-level/aliases.nix?

This wasn't done in the past and I don't see a reason to do this (assuming you mean an alias like `postgresql_11 = throw "...";`): AFAIK this mostly happens when an entire package is thrown out or when a major version is dropped not only on unstable, but everywhere including stable (one example for this is the Linux kernel).

In this case it's mostly a matter of reading the release notes or getting an error when evaluating your NixOS deployment which should be sufficient IMHO.

> What about postgresql_10?

Assuming you mean that this isn't in the `package = ...` expression: to be frank, I don't know. But if I had to guess there was no stateVersion of NixOS which defaulted to postgresql_10, so no reason to have it here. Seems out of scope to me though (unless I'm missing something).

marsam reviewed (approved):
LGTM. Thanks!

Would you mind removing [`pipelinedb.nix`](https://github.com/NixOS/nixpkgs/blob/6cf8670e843abfd5c9d176bfd634dcabad2c0237/pkgs/servers/sql/postgresql/ext/pipelinedb.nix) and other references of `"11"` in [the extensions](https://github.com/NixOS/nixpkgs/tree/6cf8670e843abfd5c9d176bfd634dcabad2c0237/pkgs/servers/sql/postgresql/ext/)?

Ma27 committed:
postgresqlPackages: remove compat with postgresql_11

marsam commented on pkgs/servers/sql/postgresql/ext/pipelinedb.nix:
```diff
@@ -34,6 +34,5 @@ stdenv.mkDerivation rec {
     license = licenses.asl20;
     platforms = postgresql.meta.platforms;
     maintainers = [ maintainers.marsam ];
-    broken = versions.major postgresql.version != "11";
```
I think it could better to remove this file, since PipelineDB only supports PostgreSQL 11, and has been abandoned by upstream.

Ma27 replied:
(not to self: `== != !=` :upside_down_face: )

Anyways, dropped it now!

Ma27 committed:
postgresql: remove pipelinedb extension

This one is apparently abandoned by upstream and only supported
postgresql_11 which is about to be dropped from nixpkgs.

Ma27 commented:
Huh just realized that there's still a reference left in `nixos/tests/postgresql.nix`. Will fix that as well.

Ma27 committed:
nixos/postgresql: remove postgresql_11-backup-all test

I don't really understand why this is only for a single version, so I
figured I'd just add this to the test matrix to also cover this
test-case for each major. Now, there's also one thing less to take care of
when removing old postgresql versions.

marsam reviewed (approved):
Thank you!

rvl commented:
I was upgrading my configuration from 23.05 to 23.11 and got this error:

```
error: postgresql_11 was removed, please upgrade your postgresql version.
```

It was confusing because I have never set `services.postgresql.package = pkgs.postgresql_11`. Turns out it was implicitly set through `system.stateVersion = "20.03"`.

@ma27, I suggest changing the error message to something like this:
```
mkThrow = ver: throw ''
  Your setting of system.stateVersion = "${config.system.stateVersion}"
  implies using pkgs.postgresql_${ver}.
  However, this version was removed. Please upgrade your postgresql version by setting
  services.postgresql.package = pkgs.postgresql_13
  (or any later version).
'';
```

Ma27 commented:
Feel free to file a PR for that.

rvl commented:
PR #272834 filed, @Ma27
